calendar_resource/models/calendar_event.py

```diff
@@ -16,11 +18,89 @@ class CalendarEvent(models.Model):
         comodel_name='resource.resource',
     )
 
-    @api.constrains('resource_ids')
-    def _check_resource_ids_dbl_book(self):
+    @api.model
+    def _format_datetime_interval_list(self, intervals):
+        """ Converts intervals to string values for front-end
+            display purposes, taking into account language,
+            timezone, as well as date and time formats.
+
+        Example:
+
+            .. code-block python
+
+            self._format_datetime_interval_list(
+                intervals=[
+                    (
+                        datetime(2017, 6, 28, 17, 0, 0),
+                        datetime(2017, 6, 29, 8, 0, 0),
+                    ),
+                    (
+                        datetime(2017, 6, 29, 12, 0, 0),
+                        datetime(2017, 6, 29, 13, 0, 0),
+                    ),
+                ],
+            )
+
+            The above example will output the following string,
+            depending on date and time format, as well as timezone.
+
+            \n06/28/2017 at 17:00:00 To\n
+            06/29/2017 at 08:00:00 (UTC)\n
+
+            \n06/29/2017 at 12:00:00 To\n
+            06/29/2017 at 13:00:00 (UTC)\n
+
+        """
+        datetimes = ''
```

Comment:
This is correct as a string, however I think the method should be renamed as it's misleading about what's being returned.